### PR TITLE
[Fix] Fixed handle_link_down to deactivate interface

### LINKED ISSUE #35: After shutting an interface down in the switch it still stay `active` from the topology perspective
I've stumbled upon this when I was resolving issue #34, it turns out it's another problem related to how the active state is being set. The expected behavior is, if an interface is shut down, the switch will send an OpenFlow event, `of_core` will publish accordingly, and then the `topology` should be setting the interface as inactive and if any link that it belongs to inactive. 

## How to reproduce

1 - start kytos `kytosd -f -E`
2 - start mininet `sudo mn -c && sudo mn --controller=remote,ip=127.0.0.1,port=6653 --switch=ovsk,protocols=OpenFlow13 --topo=linear,3`
3 - make a rest request to get the topology `http localhost:8181/api/kytos/topology/v3/interfaces`, notice the interfaces are active as expected, once the switches complete the OF handshake
4 - shutdown an nni interface `sudo ifconfig s1-eth2 down`
5 - notice this interface will stay as `active` 

here's a diff before and after the step number 4, notice the only diff are the stat counters: 

```
❯ diff --unified before.json after.json
--- before.json 2021-12-22 10:33:32.156855001 -0300
+++ after.json  2021-12-22 10:34:44.253531066 -0300
@@ -14,17 +14,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 726,
+                "rx_bytes": 796,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 9,
-                "tx_bytes": 2922,
+                "rx_packets": 10,
+                "tx_bytes": 3714,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 37
+                "tx_packets": 57
             },
             "switch": "00:00:00:00:00:00:00:01",
             "type": "interface",
@@ -44,17 +44,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 2992,
+                "rx_bytes": 3784,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 38,
-                "tx_bytes": 2992,
+                "rx_packets": 58,
+                "tx_bytes": 3714,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 38
+                "tx_packets": 57
             },
             "switch": "00:00:00:00:00:00:00:01",
             "type": "interface",
@@ -104,17 +104,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 796,
+                "rx_bytes": 866,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 10,
-                "tx_bytes": 2992,
+                "rx_packets": 11,
+                "tx_bytes": 3784,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 38
+                "tx_packets": 58
             },
             "switch": "00:00:00:00:00:00:00:02",
             "type": "interface",
@@ -134,17 +134,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 2992,
+                "rx_bytes": 3714,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 38,
-                "tx_bytes": 2992,
+                "rx_packets": 57,
+                "tx_bytes": 3784,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 38
+                "tx_packets": 58
             },
             "switch": "00:00:00:00:00:00:00:02",
             "type": "interface",
@@ -164,17 +164,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 2922,
+                "rx_bytes": 3714,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 37,
-                "tx_bytes": 2992,
+                "rx_packets": 57,
+                "tx_bytes": 3784,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 38
+                "tx_packets": 58
             },
             "switch": "00:00:00:00:00:00:00:02",
             "type": "interface",
@@ -224,17 +224,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 726,
+                "rx_bytes": 796,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 9,
-                "tx_bytes": 2922,
+                "rx_packets": 10,
+                "tx_bytes": 3714,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 37
+                "tx_packets": 57
             },
             "switch": "00:00:00:00:00:00:00:03",
             "type": "interface",
@@ -254,17 +254,17 @@
             "speed": 1250000000.0,
             "stats": {
                 "collisions": 0,
-                "rx_bytes": 2992,
+                "rx_bytes": 3784,
                 "rx_crc_err": 0,
                 "rx_dropped": 0,
                 "rx_errors": 0,
                 "rx_frame_err": 0,
                 "rx_over_err": 0,
-                "rx_packets": 38,
-                "tx_bytes": 2922,
+                "rx_packets": 58,
+                "tx_bytes": 3714,
                 "tx_dropped": 0,
                 "tx_errors": 0,
-                "tx_packets": 37
+                "tx_packets": 57
             },
             "switch": "00:00:00:00:00:00:00:03",
             "type": "interface",
```

### PR DESCRIPTION
**Pull request by viniarck**
Fixes #35 

### Release notes

```
[3.8.0] - 2021-12-22
```

#### Changed

- Fixed ``handle_link_down`` to also deactivate the interface

Now the same diff as presented in the original issue is correct, notice that `00:00:00:00:00:00:00:02:2"` is also down, because it's the other end of the veth pair see the `LOWERLAYERDOWN` below for more info: 

```
--- before.json 2021-12-22 11:38:02.277432565 -0300
+++ after.json  2021-12-22 11:38:20.427448498 -0300
@@ -68,10 +68,10 @@
                 "metadata": {}
             },
             "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260": {
-                "active": true,
+                "active": false,
                 "enabled": true,
                 "endpoint_a": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:02:2",
                     "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
@@ -101,7 +101,7 @@
                     "uni": false
                 },
                 "endpoint_b": {
-                    "active": true,
+                    "active": false,
                     "enabled": true,
                     "id": "00:00:00:00:00:00:00:01:2",
                     "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
@@ -175,7 +175,7 @@
                         "uni": true
                     },
                     "00:00:00:00:00:00:00:01:2": {
-                        "active": true,
+                        "active": false,
                         "enabled": true,
                         "id": "00:00:00:00:00:00:00:01:2",
                         "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",
@@ -286,7 +286,7 @@
                         "uni": true
                     },
                     "00:00:00:00:00:00:00:02:2": {
-                        "active": true,
+                        "active": false,
                         "enabled": true,
                         "id": "00:00:00:00:00:00:00:02:2",
                         "link": "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260",

```

```
❯ sudo ip link show dev s1-eth2
43: s1-eth2@s2-eth2: <BROADCAST,MULTICAST> mtu 1500 qdisc noqueue master ovs-system state DOWN mode DEFAULT group default qlen 1000
    link/ether c2:41:ba:72:f9:db brd ff:ff:ff:ff:ff:ff

❯ sudo ip link show dev s2-eth2
44: s2-eth2@s1-eth2: <NO-CARRIER,BROADCAST,MULTICAST,UP,M-DOWN> mtu 1500 qdisc noqueue master ovs-system state LOWERLAYERDOWN mode DEFAULT group default qlen 1000
    link/ether ae:12:78:fd:88:fb brd ff:ff:ff:ff:ff:ff

```

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -680,6 +680,7 @@ def handle_switch_maintenance_start(self, event):
 
     def handle_link_down(self, interface):
         """Notify a link is down."""
+        interface.deactivate()
```
Comment (italovalcy) [inline main.py:683]: As we discussed, by deactivating the interface here, the link will automatically become `active=False` (as per https://github.com/kytos-ng/kytos/blob/master/kytos/core/link.py#L60-L61) and no event `kytos/topology.link_down` will be generated (which impacts mef_eline and others). Maybe we can move the `interface.deactivate()` down to line 689 (after `notify_link_status_change()`, outside of the `if`)

**Review by italovalcy — CHANGES_REQUESTED** [review]
Looks great to me! Thanks for finding and fixing this issue so quickly, Vinicius! I just left a comment regarding the location where the interface is being deactivated and the risk of not generating link_down events.

File: main.py (hunk 1 above)
Comment (viniarck) [reply main.py]: done https://github.com/kytos-ng/topology/pull/37/commits/9d1a54f83f077a9a0f17266802baf517a9863e4e

Comment (italovalcy) [reply main.py]: Checked and it looks good to me!

**Review by italovalcy — APPROVED** [review]

**Comment by viniarck** [comment]
Appreciated your review @italovalcy, and thanks for catching the issue with the notification that was also fixed.

**Pull request merged**